packages/block-library/src/latest-comments/deprecated.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * WordPress dependencies
+ */
+
+const deprecated = [
+	{
+		// Run this deprecation only if the legacy attribute exists.
+		isEligible( attributes ) {
+			return (
+				typeof attributes !== 'undefined' &&
+				typeof attributes.displayExcerpt !== 'undefined'
+			);
+		},
+
+		attributes: {
+			commentsToShow: {
+				type: 'number',
+				default: 5,
+			},
+			displayAvatar: {
+				type: 'boolean',
+				default: true,
+			},
+			displayDate: {
+				type: 'boolean',
+				default: true,
+			},
+			// Legacy attribute (deprecated)
+			displayExcerpt: {
+				type: 'boolean',
+				default: true,
+			},
+		},
+
+		/**
+		 * Migrates legacy block attributes to the new format.
+		 *
+		 * @param {Object} attributes
+		 * @return {Object}
+		 */
+		migrate( attributes ) {
+			const commentsToShow =
+				typeof attributes.commentsToShow !== 'undefined'
+					? attributes.commentsToShow
+					: 5;
+			const displayAvatar =
+				typeof attributes.displayAvatar !== 'undefined'
+					? attributes.displayAvatar
+					: true;
+			const displayDate =
+				typeof attributes.displayDate !== 'undefined'
+					? attributes.displayDate
+					: true;
+
+			const displayContent = attributes.displayExcerpt
+				? 'excerpt'
+				: 'none';
+
+			try {
+				console.log(
+					'latest-comments: deprecation migrate() ran — displayContent:',
+					displayContent
+				);
+			} catch ( e ) {
+				// ignore
+			}
+
+			return {
+				commentsToShow,
+				displayAvatar,
+				displayDate,
+				displayContent,
+			};
+		},
+
+		// Dynamic block → no client-side save.
+		save() {
+			return null;
+		},
```

Comment:
```suggestion
		attributes: {
			commentsToShow: {
				type: 'number',
				default: 5,
			},
			displayAvatar: {
				type: 'boolean',
				default: true,
			},
			displayDate: {
				type: 'boolean',
				default: true,
			},
			displayExcerpt: {
				type: 'boolean',
				default: true,
			},
		},
		isEligible( attributes ) {
			return attributes?.displayExcerpt === false;
		},
		migrate( attributes ) {
			return {
				...attributes,
				displayContent: attributes.displayExcerpt ? 'excerpt' : 'none',
			};
		},
```

It should be possible to write it more simply.